model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/GeneratorInput.kt

```diff
@@ -253,6 +278,17 @@ internal class ProcessedConcept(
     fun getAllSuperConcepts(): Sequence<ProcessedConcept> = getAllSuperConcepts_().distinct()
     fun getAllSuperConceptsAndSelf(): Sequence<ProcessedConcept> = getAllSuperConceptsAndSelf_().distinct()
     fun getDuplicateSuperConcepts() = getAllSuperConcepts_().groupBy { it }.filter { it.value.size > 1 }.map { it.key }
+
+    fun getNonInterfaceSuperConcept() = getDirectSuperConcepts().firstOrNull()
```

Comment:
There are no "interface concepts" in Modelix. This is a term from MPS, but the generator shouldn't make any MPS specific assumptions. But after replacing `conceptFqName2RootMetaProperties` with a single Set, these method shouldn't be required anymore.